iree/compiler/Translation/IREEVM.cpp

```diff
@@ -59,12 +59,16 @@ static InputDialectOptions getInputDialectOptionsFromFlags() {
   static llvm::cl::opt<InputDialectOptions::Type> *typeFlag =
       new llvm::cl::opt<InputDialectOptions::Type>{
           "iree-input-type", llvm::cl::desc("IREE input type"),
-          llvm::cl::values(clEnumValN(InputDialectOptions::Type::none, "none",
-                                      "No input dialect transformation"),
-                           clEnumValN(InputDialectOptions::Type::tosa, "tosa",
-                                      "Legalize from TOSA ops"),
-                           clEnumValN(InputDialectOptions::Type::mhlo, "mhlo",
-                                      "Legalize from MHLO ops")),
+          llvm::cl::values(
+              clEnumValN(InputDialectOptions::Type::none, "none",
+                         "No input dialect transformation"),
+              clEnumValN(InputDialectOptions::Type::tosa, "tosa",
+                         "Legalize from TOSA ops"),
+              clEnumValN(InputDialectOptions::Type::mhlo, "mhlo",
+                         "Legalize from MHLO ops"),
+              clEnumValN(
+                  InputDialectOptions::Type::xla, "xla",
+                  "Legalize from MHLO ops (with XLA cleanup preprocessing)")),
```

Comment:
Is it worth having these two separate? They're both the same dialect. (How much) does it hurt to run this always for MHLO input?

Reply:
I am still (perhaps in vain) trying to keep things sequestered that *force* us into CFG, which this does. Those are all things we need to clean up eventually, and at least keeping this as its own thing keeps us from making it load bearing for most uses of the compiler.